docs/source/Diff.rst

```diff
@@ -0,0 +1,92 @@
+.. _job_diff_:
+
+========
+Job Diff
+========
+
+Avocado Diff plugin allows users to easily compare several aspects of
+two given jobs. The basic usage is::
+
+    $ avocado diff 7025aaba 384b949c
+    --- 7025aaba9c2ab8b4bba2e33b64db3824810bb5df
+    +++ 384b949c991b8ab324ce67c9d9ba761fd07672ff
+    @@ -1,15 +1,15 @@
+     
+     COMMAND LINE
+    -/usr/bin/avocado run sleeptest.py
+    +/usr/bin/avocado run passtest.py
+     
+     TOTAL TIME
+    -1.00 s
+    +0.00 s
+     
+     TEST RESULTS
+    -1-sleeptest.py:SleepTest.test: PASS
+    +1-passtest.py:PassTest.test: PASS
+     
+     ...
+
+Avocado Diff can compare and create an unified diff of:
+
+- Command line.
+- Job time.
+- Variants and parameters.
+- Tests results.
+- Configuration.
+- Sysinfo pre and post.
+
+Only sections with different content will be included in the results. You
+can also enable/disable those sections with ``--diff-filter``. Please see
+``avocado diff --help`` for more information.
+
+Jobs can be identified by the Job ID, by the results directory or by the
+key ``latest``. Example::
+
+    $ avocado diff ~/avocado/job-results/job-2016-08-03T15.56-4b3cb5b/ latest
+    --- 4b3cb5bbbb2435c91c7b557eebc09997d4a0f544
+    +++ 57e5bbb3991718b216d787848171b446f60b3262
+    @@ -1,9 +1,9 @@
+
+     COMMAND LINE
+    -/usr/bin/avocado run perfmon.py
+    +/usr/bin/avocado run passtest.py
+
+     TOTAL TIME
+    -11.91 s
+    +0.00 s
+
+     TEST RESULTS
+    -1-test.py:Perfmon.test: FAIL
+    +1-examples/tests/passtest.py:PassTest.test: PASS
+
+
+
+Along with the unified diff, you can also generate the html (option ``--html``)
+diff file and, optionally, open it on your preferred browser (option
+``--open-browser``)::
```

Comment:
On Wed, Aug 10, 2016 at 06:26:00AM -0700, Lukáš Doktor wrote:
> > +    -/usr/bin/avocado run perfmon.py
> > +    +/usr/bin/avocado run passtest.py
> > +
> > +     TOTAL TIME
> > +    -11.91 s
> > +    +0.00 s
> > +
> > +     TEST RESULTS
> > +    -1-test.py:Perfmon.test: FAIL
> > +    +1-examples/tests/passtest.py:PassTest.test: PASS
> > +
> > +
> > +
> > +Along with the unified diff, you can also generate the html (option ``--html``)
> > +diff file and, optionally, open it on your preferred browser (option
> > +``--open-browser``)::
> 
> Well the `-html` version is unusable on my machine as the
> `sysinfo` lines are too long. I'd actually rather see an option
> to store the 2 source files in order to be able to view it in
> my favorite diff tool (meld).
> 

This is something I suggested in the past, although in a
different form: I think the reports with the job summary are
quite useful, independent of the diff feature. So my suggestion
is that we start including a job-summary.txt report as part of
the test results.

The job diff feature will still be quite useful (--diff-filter,
for example, is quite nice), but much simpler to implement.

BUT: the job feature should be delivered this sprint, so let's
leave improvements like that to a future to-do. The most
important thing is that we get the UI right (command line options
and basic output).

-- 
Ademar Reis
Red Hat

^[:wq!